### ISSUE TITLE
Add back StatusDescription to HttpStatusCodeResult

### ISSUE THREAD

**Issue by jods4** [aspnet/Mvc#3455]
I think everything is in the title. 
Previously in MVC HttpStatusCodeResult allowed users to customize the http status message returned to client.

**Comment by Eilon** [comment]
@Tratcher I think you've had thoughts on this before, and concluded that it was not a good idea. Do you recall the reason?

**Comment by Tratcher** [comment]
Won't fix. https://github.com/aspnet/HttpAbstractions/issues/395

**Comment by Eilon** [comment]
Done, thanks.

**Comment by jods4** [comment]
But... David Fowler said:
> Closing this out until somebody asks for it.

Just joking. I did not know it was removed from HTTP/2, which is a good reason to obsolete the feature. Also, there is a workaround available in the linked issue so it's not like it's undoable.

BTW my use case was that I was used to signal "expected" errors in a web api (e.g. failed validation) by returning an error http status with friendly explanation in the reason phrase (which was picked up by app and shown to user). But I can do pretty much the same thing using the response body anyway, especially now that I know HTTP/2 removed this.
